dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/ParameterCount/ParameterCountAnalyzer.cs

```diff
@@ -145,21 +157,35 @@ private static string GetNameForMethod(IMethodSymbol method)
             return builder.ToString();
         }
 
-        private static void AnalyzeParameters(SymbolAnalysisContext context, ParameterCountInfo parameterCountInfo, string memberName)
+        private static void AnalyzeParameters(SymbolAnalysisContext context, ParameterCountInfo parameterCountInfo,
+            string memberName)
         {
-            ImmutableArray<IParameterSymbol> parameters = parameterCountInfo.MethodSymbol.Parameters;
+            var methodParameters = parameterCountInfo.MethodSymbol.Parameters;
 
-            if (parameters.Length > parameterCountInfo.MaxParameterCount)
+            var parametersWithoutExcludedTypes = methodParameters.AsEnumerable()
+                .Where(parameter =>
+                {
+                    var parameterTypeName = parameter.Type.Name;
+
+                    return !parameter.IsLastParameter(methodParameters) ||
```

Comment:
I'm not sure if this is the correct logic. Don't we just want to exclude parameter types? What if we want to add a type that might not be the last parameter? I think there's an MS analyzer that suggests `CancelationToken` should be the last parameter so we can maybe assume that will always be the case.